pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -155,3 +189,55 @@ func (nc *nodeConfig) handleCSR(requestorFilter string) error {
 
 	return nil
 }
+
+// GetNode returns a pointer to the node object associated with the external IP provided
+func (nc *nodeConfig) getNode(externalIP string) (*v1.Node, error) {
+	var matchedNode *v1.Node
+
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get list of nodes")
+	}
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found")
+	}
+
+	// Find the node that has the given IP
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "ExternalIP" && address.Address == externalIP {
+				matchedNode = &node
+				break
+			}
+		}
+		if matchedNode != nil {
+			break
+		}
+	}
+	if matchedNode == nil {
+		return nil, fmt.Errorf("could not find node with IP: %s", externalIP)
+	}
+	return matchedNode, nil
+}
+
+// waitForHybridOverlayAnnotation waits for the hybrid overlay subnet annotation to be present on the node until the
+// timeout is reached
+func (nc *nodeConfig) waitForHybridOverlayAnnotation(nodeName string) error {
```

Comment:
This function should use `func (tc *testContext) waitForNodeAnnotation(nodeName, annotation string) error`
See my comments in the e2e test for more details